src/globalsearch/spotifywebapisearchprovider.cpp

```diff
@@ -0,0 +1,91 @@
+/* This file is part of Clementine.
+   Copyright 2021, Kenman Tsang <kentsangkm@pm.me>
+
+   Clementine is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   Clementine is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Clementine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "spotifywebapisearchprovider.h"
+
+#include <qurl.h>
+
+#include <iostream>
+
+#include "internet/spotifywebapi/spotifywebapiservice.h"
+#include "ui/iconloader.h"
+
+SpotifyWebApiSearchProvider::SpotifyWebApiSearchProvider(
+    Application* app, SpotifyWebApiService* parent)
+    : SearchProvider(app, parent), parent_{parent} {
+  Init("Spotify (Experimential)", "spotify_web_api",
+       IconLoader::Load("spotify", IconLoader::Provider),
+       WantsDelayedQueries | WantsSerialisedArtQueries | ArtIsProbablyRemote |
+           CanGiveSuggestions);
+
+  connect(parent, &SpotifyWebApiService::SearchFinished, this,
+          &SpotifyWebApiSearchProvider::SearchFinishedSlot);
+}
+
+void SpotifyWebApiSearchProvider::SearchAsync(int id, const QString& query) {
+  if (lastSearchId_ != NO_RUNNING_SEARCH) {
+    // Cancel last pending search
+    emit SearchFinished(lastSearchId_);
+
+    // Set the pending query
+    lastSearchId_ = id;
```

Comment:
Variables names should all be `snake_case_`. We should figure out if we can make clang-format enforce that.